DSharpPlus.UnifiedCommands/Message/Internals/MessageHandler.cs

```diff
@@ -0,0 +1,188 @@
+                        new InvalidMessageConversionError
+                        {
+                            Name = e.Name,
+                            IsPositionalArgument = e.IsPositionalArgument,
+                            Value = e.Value,
+                            Type = e.Type,
+                        }
+                        , _message);
+                    return;
+                }
+            }
+
+            if (_conditionBuilders.Count != 0)
+            {
+                MessageConditionHandler conditionHandler =
+                    new(_conditionBuilders);
+                bool shouldContinue =
+                    await conditionHandler.StartGoingThroughConditionsAsync(
+                        new MessageContext { Message = _message, Data = new(_name, _data.Method), Client = _client },
+                        _scope);
+                if (!shouldContinue)
+                {
+                    return;
+                }
+            }
+
+            TimeSpan elapsedTime = Stopwatch.GetElapsedTime(startTime);
+            _client.Logger.LogDebug("Took {NsExecution}ns to process everything", elapsedTime.TotalNanoseconds);
+
+            try
+            {
+                if (_data.ReturnsNothing)
+                {
+                    if (_data.IsAsync)
+                    {
+                        await (Task)_data.Method.Invoke(_module, BindingFlags.OptionalParamBinding,
+                            null, parameters, null)!;
+                    }
+                    else
+                    {
+                        _data.Method.Invoke(_module, BindingFlags.OptionalParamBinding, null, parameters, null);
+                    }
+                }
+                else
+                {
+                    if (_data.IsAsync)
+                    {
+                        Task<IMessageResult> task =
+                            (Task<IMessageResult>)_data.Method.Invoke(_module,
+                                BindingFlags.OptionalParamBinding, null,
+                                parameters, null)!;
+                        IMessageResult result = await task;
+                        await TurnResultIntoActionAsync(result);
+                    }
+                    else
+                    {
+                        IMessageResult result =
+                            (IMessageResult)_data.Method.Invoke(_module, parameters)!;
+                        await TurnResultIntoActionAsync(result);
+                    }
+                }
+            }
+            catch (TargetInvocationException e)
+            {
+                System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(e.InnerException ?? e).Throw();
+                throw e.InnerException;
+            }
+            catch (AggregateException e)
+            {
+                System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(e.InnerException ?? e).Throw();
```

Comment:
`using System.Runtime.ExceptionServices;`
also, port the documentation over from application commands